### PR TITLE
Add ssh_user option to openstack_taster.

### PR DESCRIPTION
**Pull request by Codysseus**
#50

### CODE REVIEW

File: bin/openstack_taster [hunk 1]
```diff
@@ -66,8 +64,11 @@ rescue StandardError => e
   exit(1)
 end
 
+image_name = params[0]
+settings[:ssh_user] = params[1]
+
 require 'fog/openstack'
-require 'openstack_taster'
+require './lib/openstack_taster'
```
Comment (Codysseus) [inline bin/openstack_taster]: I added this bit so testing would work locally.

Comment (ramereth) [reply bin/openstack_taster]: Could this just be ``'lib/openstack_taster'`` instead of the leading ``./``?

Comment (Codysseus) [reply bin/openstack_taster]: I tried to do that when testing locally, but that didn't work either. The current directory is not in the `$LOAD_PATH` variable.

Comment (ramereth) [reply bin/openstack_taster]: I'm confused. Why does this need to be added in the first place? Have you tried testing it via building the gem and installing it?

Comment (Codysseus) [reply bin/openstack_taster]: I was testing this by running `ruby bin/openstack_taster`. I forgot about using the `-I` flag.

**Comment by Codysseus** [comment]
@ramereth Fixed require statement. You should be able to test normally by installing the gem.

File: bin/openstack_taster [hunk 2]
```diff
@@ -12,8 +12,8 @@ settings = {}
 ARGV << '-h' if ARGV.empty?
 
 parser = OptionParser.new do |opts|
-  opts.banner = "Usage: openstack_taster <image_name> {suite_name} [--create-snapshot]\n" \
-                "   or: openstack_taster <image_name> [--create-snapshot]\n" \
+  opts.banner = "Usage: openstack_taster <image_name> <ssh_user> {suite_name}  [--create-snapshot]\n" \
+                "   or: openstack_taster <image_name> <ssh_user> [--create-snapshot]\n" \
                 '   or: openstack_taster <option>'
```
Comment (TravisWhitehead) [inline bin/openstack_taster:17]: What is `openstack_taster <option>` for? Below it appears you removed the case where `params.length == 1`, suggesting this may no longer be a valid usage?

File: lib/openstack_taster.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@
 # @author Andrew Tolvstad, Samarendra Hedaoo, Cody Holliday
 class OpenStackTaster
   INSTANCE_FLAVOR_NAME = 'm1.tiny'
-  INSTANCE_NETWORK_NAME = 'public'
+  INSTANCE_NETWORK_NAME = 'public1'
```
Comment (TravisWhitehead) [inline lib/openstack_taster.rb:13]: Do we only intend to use `openstack_taster` on https://oprod-controller1.osuosl.org/ (or OpenStack instances where we have a network named `public1`)?

Perhaps this would be better off as a configurable value? (Not necessarily recommending that should be a part of this PR.)

**Review by TravisWhitehead — CHANGES_REQUESTED** [review]

File: lib/openstack_taster.rb (hunk 3 above)
Comment (ramereth) [reply lib/openstack_taster.rb:13]: This needs to be configurable per cluster and should be made in a separate PR to fix it.

File: bin/openstack_taster (hunk 2 above)
Comment (Codysseus) [reply bin/openstack_taster:17]: The only valid option is the `-h` option, which causes the program to quit with a usage message before the case statement.

**Comment by Codysseus** [comment]
@TravisWhitehead @ramereth Any other comments or reviews?

File: Gemfile [hunk 4]
```diff
@@ -4,5 +4,5 @@ source 'https://rubygems.org' do
   gem 'fog'
   gem 'inspec'
   gem 'net-ssh'
-  gem 'optparse'
+  gem 'rubysl-optparse'
```
Comment (ramereth) [inline Gemfile:7]: Why this change?

**Review by ramereth — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 4 above)
Comment (Codysseus) [reply Gemfile:7]: They changed the name recently. `optparse` no longer exists as a package.

**Review by TravisWhitehead — APPROVED** [review]

**Review by ramereth — APPROVED** [review]
I just noticed that this now requires a username to be given to get this to work. We'll need to update stuff down stream to account for that but I think that's a better way to go around this anyway.
